test_communication/CMakeLists.txt

```diff
@@ -113,126 +113,136 @@ if(BUILD_TESTING)
   macro(multi_targets)
     # test publish / subscribe messages
     if(rmw_implementation1 STREQUAL rmw_implementation2)
-      set(suffix "__${rmw_implementation1}")
+      set(suffix "${suffix}__${rmw_implementation1}")
     else()
-      set(suffix "__${rmw_implementation1}__${rmw_implementation2}")
+      set(suffix "${suffix}__${rmw_implementation1}__${rmw_implementation2}")
     endif()
-    set(TEST_PUBLISHER_EXECUTABLE "$<TARGET_FILE:test_publisher_cpp__${rmw_implementation1}>")
     set(PUBLISHER_RMW ${rmw_implementation1})
-    set(TEST_SUBSCRIBER_EXECUTABLE "$<TARGET_FILE:test_subscriber_cpp__${rmw_implementation2}>")
     set(SUBSCRIBER_RMW ${rmw_implementation2})
     foreach(message_file ${message_files})
       get_filename_component(TEST_MESSAGE_TYPE "${message_file}" NAME_WE)
       set(test_suffix "__${TEST_MESSAGE_TYPE}${suffix}")
       configure_file(
         test/test_publisher_subscriber.py.in
-        test_publisher_subscriber_cpp${test_suffix}.py.configured
+        test_publisher_subscriber${test_suffix}.py.configured
         @ONLY
       )
       file(GENERATE
-        OUTPUT "test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py"
-        INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}.py.configured"
+        OUTPUT "test_publisher_subscriber${test_suffix}_$<CONFIGURATION>.py"
+        INPUT "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber${test_suffix}.py.configured"
       )
 
       # TODO(dirk-thomas) publishing Bounded/DynamicArrayPrimitives in OpenSplice is buggy
       # https://github.com/ros2/rclcpp/issues/219
-      if((NOT TEST_MESSAGE_TYPE STREQUAL "BoundedArrayPrimitives" AND NOT TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives") OR NOT rmw_implementation1 STREQUAL "rmw_opensplice_cpp")
-        ament_add_nose_test(
-          publisher_subscriber_cpp${test_suffix}
-          "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber_cpp${test_suffix}_$<CONFIGURATION>.py"
+      if(
+        (NOT TEST_MESSAGE_TYPE STREQUAL "BoundedArrayPrimitives" AND NOT TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives") OR
+        NOT rmw_implementation1 STREQUAL "rmw_opensplice_cpp"
+      )
+        # TODO(mikaelarguedas) Simpler way to blacklist specific tests (e.g. regex matching)
+        ament_add_nose_test(test_publisher_subscriber${test_suffix}
+          "${CMAKE_CURRENT_BINARY_DIR}/test_publisher_subscriber${test_suffix}_$<CONFIGURATION>.py"
+          WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py"
+          PYTHON_EXECUTABLE "${_PYTHON_EXECUTABLE}"
+          APPEND_ENV PYTHONPATH="${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py:${CMAKE_CURRENT_SOURCE_DIR}:${CMAKE_CURRENT_BINARY_DIR}/../../rclpy"
+          APPEND_ENV AMENT_PREFIX_PATH=${ament_index_build_path}
```

Comment:
it is not, that looks like an artifact of a copy/paste from the `rclpy`'s CMakeLists when I first added the python tests to this package